genesyscloud/resource_genesyscloud_outbound_dnclist.go

```diff
@@ -0,0 +1,376 @@
+	return resources, nil
+}
+
+func outboundDncListExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllOutboundDncLists),
+		RefAttrs: map[string]*RefAttrSettings{
+			"division_id": {RefType: "genesyscloud_auth_division"},
+		},
+	}
+}
+
+func resourceOutboundDncList() *schema.Resource {
+	return &schema.Resource{
+		Description: `Genesys Cloud Outbound DNC List`,
+
+		CreateContext: createWithPooledClient(createOutboundDncList),
+		ReadContext:   readWithPooledClient(readOutboundDncList),
+		UpdateContext: updateWithPooledClient(updateOutboundDncList),
+		DeleteContext: deleteWithPooledClient(deleteOutboundDncList),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: `The name of the DncList.`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`contact_method`: {
+				Description:  `The contact method. Required if dncSourceType is rds.`,
+				Optional:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{`Email`, `Phone`}, false),
+			},
+			`login_id`: {
+				Description: `A dnc.com loginId. Required if the dncSourceType is dnc.com.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`campaign_id`: {
+				Description: `A dnc.com campaignId. Optional if the dncSourceType is dnc.com.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`dnc_codes`: {
+				Description: `The list of dnc.com codes to be treated as DNC. Required if the dncSourceType is dnc.com.`,
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{`B`, `C`, `D`, `E`, `F`, `G`, `H`, `I`, `L`, `M`, `O`, `P`, `R`, `S`, `T`, `V`, `W`, `X`, `Y`}, false),
+				},
+			},
+			`license_id`: {
+				Description: `A gryphon license number. Required if the dncSourceType is gryphon.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`division_id`: {
+				Description: `The division this DNC List belongs to.`,
+				Optional:    true,
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			`dnc_source_type`: {
+				Description:  `The type of the DNC List.`,
+				Required:     true,
+				ForceNew:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{`rds`, `dnc.com`, `gryphon`}, false),
+			},
+			`entries`: {
+				Description: `Rows to add to the DNC list. To emulate removing phone numbers, you can set expiration_date to a date in the past.`,
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						`expiration_date`: {
+							Description: `Expiration date for DNC phone numbers in yyyy-MM-ddTHH:mmZ format.`,
```

Comment:
Would be good to have `ValidateDiagFunc: validateLocalDateTimes` or a similar validation function if this is a different date format